streams/src/main/java/org/apache/kafka/streams/kstream/internals/FullChangeSerde.java

```diff
@@ -47,98 +43,73 @@ private FullChangeSerde(final Serde<T> inner) {
-                final int oldSize = oldBytes == null ? -1 : oldBytes.length;
-                final byte[] newBytes = data.newValue == null ? null : innerSerializer.serialize(topic, data.newValue);
-                final int newSize = newBytes == null ? -1 : newBytes.length;
-
-                final ByteBuffer buffer = ByteBuffer.wrap(
-                    new byte[4 + (oldSize == -1 ? 0 : oldSize) + 4 + (newSize == -1 ? 0 : newSize)]
-                );
-                buffer.putInt(oldSize);
-                if (oldBytes != null) {
-                    buffer.put(oldBytes);
-                }
-                buffer.putInt(newSize);
-                if (newBytes != null) {
-                    buffer.put(newBytes);
-                }
-                return buffer.array();
-            }
-
-            @Override
-            public void close() {
-                innerSerializer.close();
-            }
-        };
+    public Change<T> deserializeParts(final String topic, final Change<byte[]> serialChange) {
+        if (serialChange == null) {
+            return null;
+        }
+        final Deserializer<T> innerDeserializer = innerSerde().deserializer();
+
+        final T oldValue =
+            serialChange.oldValue == null ? null : innerDeserializer.deserialize(topic, serialChange.oldValue);
+        final T newValue =
+            serialChange.newValue == null ? null : innerDeserializer.deserialize(topic, serialChange.newValue);
+
+        return new Change<>(newValue, oldValue);
     }
 
-    @Override
-    public Deserializer<Change<T>> deserializer() {
-        final Deserializer<T> innerDeserializer = inner.deserializer();
-        return new Deserializer<Change<T>>() {
-            @Override
-            public void configure(final Map<String, ?> configs, final boolean isKey) {
-                innerDeserializer.configure(configs, isKey);
-            }
-
-            @Override
-            public Change<T> deserialize(final String topic, final byte[] data) {
-                if (data == null) {
-                    return null;
-                }
-                final ByteBuffer buffer = ByteBuffer.wrap(data);
-
-                final byte[] oldBytes = extractOldValuePart(buffer);
-                final T oldValue = oldBytes == null ? null : innerDeserializer.deserialize(topic, oldBytes);
-
-                final int newSize = buffer.getInt();
-                final byte[] newBytes = newSize == -1 ? null : new byte[newSize];
-                if (newBytes != null) {
-                    buffer.get(newBytes);
-                }
-                final T newValue = newBytes == null ? null : innerDeserializer.deserialize(topic, newBytes);
-                return new Change<>(newValue, oldValue);
-            }
-
-            @Override
-            public void close() {
-                innerDeserializer.close();
-            }
-        };
+    public static byte[] composeLegacyFormat(final Change<byte[]> serialChange) {
```

Comment:
I'm not sure what the best name for this method would be, but can we come up with something a bit more descriptive here?

Reply:
Yeah, I'm not sure either... any ideas?

Reply:
I couldn't think of anything more context-free, so I settled for adding javadocs on these methods.

Reply:
How about `mergeToSingleArray` ?